libraries/AP_BoardConfig/AP_BoardConfig.cpp

```diff
@@ -31,7 +32,11 @@
 #endif
 
 #if CONFIG_HAL_BOARD == HAL_BOARD_PX4
-# define BOARD_SAFETY_ENABLE_DEFAULT 1
+# if APM_BUILD_TYPE(APM_BUILD_ArduSub)
+#  define BOARD_SAFETY_ENABLE_DEFAULT 0
+# else
+#  define BOARD_SAFETY_ENABLE_DEFAULT 1
```

Comment:
also, note that you can do this in the ArduPub/Parameters.cpp code:
  AP_Param::set_default_by_name("BRD_SAFETYENABLE", 0);
you can use that method to change any default on a per-vehicle basis without having to change the library.